packages/service-provider-core/src/uri-generator.ts

```diff
@@ -142,7 +147,61 @@ function generateUri(options: CliOptions): string {
     validateConflicts(options);
   }
 
-  return `${Scheme.Mongo}${host || generateHost(options)}:${port || generatePort(options)}/${db || DEFAULT_DB}`;
+  return `${Scheme.Mongo}${host || generateHost(options)}:${port || generatePort(options)}${getDbAndQueryStringWithDirectConnectionIfRequired(dbAndQueryString || DEFAULT_DB)}`;
+}
+
+/**
+ * Parses a given mongodb:// connection string and adds the `directConnection=true` query parameter if required.
+ * See: https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#reference-implementation
+ * @param uri mongodb:// connection string
+ */
+function addDirectConnectionQueryParameterToMongoUriIfRequired(uri: string): string {
+  const uriNoScheme = uri.substr(Scheme.Mongo.length);
+
+  // Split URI at first "/"
```

Comment:
Btw, I think it should be fine to use `new URL(uri)` here for parsing and modification. I’m not sure if that makes things easier or not, but it might be nice to have an more URL-parser-y thing here?

Reply:
I'll give it a try! I didn't bother using it since the connection string spec allows for multiple hosts which would be an invalid URI... but worth trying if I would just get all in the host field.

Reply:
Okay I've tried it with `new URL(uri)` - the problem indeed are the multiple hosts which will throw an exception:
```
TypeError [ERR_INVALID_URL]: Invalid URL: mongodb://192.42.42.42:27017,192.0.0.1:27018/db?readPreference=primary
```